### PR TITLE
[Release] 2.15.1

### PR DESCRIPTION
**Pull request by abelgardep**
Implements #2903 
Needs https://github.com/owncloud/android-library/pull/333

## QA

Regression test:

BUGS & IMPROVEMENTS:

- [x] (1) Login redirections https://github.com/owncloud/android/pull/2914#issuecomment-656079251 [FIXED]
- [X] (2) Password enabled in public links https://github.com/owncloud/android/pull/2914#issuecomment-656525868 [FIXED]
- [X] (3) Password not disabled https://github.com/owncloud/android/pull/2914#issuecomment-656527657 [FIXED]
- [x] (4) Failing redirection operations https://github.com/owncloud/android/pull/2914#issuecomment-658149907 [FIXED. 302 moved]
- [x] (5) OAuth2 flow broken https://github.com/owncloud/android/pull/2914#issuecomment-658268082 [WONT FIX HERE]

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 2.15.1 (July 2020)
+- Android 10 support
+- Update network libraries to more recent versions, OkHttp + Dav4Android
+- Bug fixes, including:
+  + Fix some authentication problems
+  + Add support for version.hide server setting
```
Comment (davigonz) [inline CHANGELOG.md:6]: How does this change affect users? Couldn't we explain this in a more user-friendly way?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+## 2.15.1 (July 2020)
+- Android 10 support
+- Update network libraries to more recent versions, OkHttp + Dav4Android
+- Bug fixes, including:
+  + Fix some authentication problems
```
Comment (davigonz) [inline CHANGELOG.md:5]: Could you mention those authentication problems in order to let the users with those problems know that are already fixed?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
+## 2.15.1 (July 2020)
+- Android 10 support
```
Comment (davigonz) [inline CHANGELOG.md:2]: Any interesting improvement for the users included in Android 10?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
...
+- Update network libraries to more recent versions, OkHttp + Dav4Android
```
Comment (jesmrec) [inline CHANGELOG.md:3]: We should use the current name: dav4jvm. Or using it in this way:

dav4jvm (old dav4Android)

File: CHANGELOG.md (hunk 3 above)
Comment (jesmrec) [reply CHANGELOG.md:2]: we can include that we support TLS 1.3 from now on. Something like:

`Android 10: Support to TLS 1.3`

**Comment by jesmrec** [comment]
### (1)

1. Enter a URL that redirects to other one (no matter if redirection is 301 or 302)
2. Enter credentials

Current: unknown error
Expected: login correct

Pixel2, Android10
commit `215dc75`

**Comment by jesmrec** [comment]
### (2)

This is a regression from 2.15, not 2.15.1. So, no mandatory to fix (desirable)

1. Create a public link over a folder, setting an expiration date.
2. Submit the link
3. Click on the pencil icon to edit the link

Current: Password is enabled
Expected: Password keeps disabled

Pixel2, Android10

**Comment by jesmrec** [comment]
### (3)

Related with (2), same problem with different steps:

### (2)

This is a regression from 2.15, not 2.15.1. So, no mandatory to fix (desirable)

1. Create a public link over a folder, setting a password
2. Submit the link
3. Click on the pencil icon to edit the link, and disable the password
4. Submit the link
5. Click on the pencil icon to edit the link

Current: Password is enabled
Expected: Password is disabled

Pixel2, Android10

**Comment by jesmrec** [comment]
### (4)

After fixing (1), some operations are not working fine in redirected servers:

- Rename, Copy, Move:  Returns the following error message: `Requested uri (/remote.php/dav/files/admin/test/) is out of base uri (/stable10/remote.php/dav/)` . 

- Shares: it is posible to create/edit shares from the app. But, shares created in the web or in other clients are not correctly retrieved and displayed. It is returned 404 with `Wrong path, file/folder doesn't exist`

- Upload from camera: taking a picture from the FAB button option, causes an `Unknown error` in Uploads view. PUT operation returns a 301 that is not being resolved.

**Comment by jesmrec** [comment]
### (5)

I got the following error stacktrace that made an OAuth2 account stop working (no token renewal). No specific steps, i will try to reproduce again to get a more accurated scenario. Please, take a look, just in case the trace is significative for you @abelgardep 

```

2020-07-08 12:10:01.272 1284-1518/? E/AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: SyncManager
    java.lang.IllegalArgumentException: Unknown package: com.owncloud.android
        at com.android.server.pm.Settings.wasPackageEverLaunchedLPr(Settings.java:4652)
        at com.android.server.pm.PackageManagerService$PackageManagerInternalImpl.wasPackageEverLaunched(PackageManagerService.java:26663)
        at com.android.server.content.SyncManager$SyncHandler.updateOrAddPeriodicSyncH(SyncManager.java:3468)
        at com.android.server.content.SyncManager$SyncHandler.handleSyncMessage(SyncManager.java:3077)
        at com.android.server.content.SyncManager$SyncHandler.handleMessage(SyncManager.java:3026)
        at android.os.Handler.dispatchMessage(Handler.java:112)
        at android.os.Looper.loop(Looper.java:216)
        at android.os.HandlerThread.run(HandlerThread.java:65)
2020-07-09 13:32:45.452 1019-1452/? E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-1
    Process: com.owncloud.android, PID: 1019
    org.koin.core.error.NoBeanDefFoundException: No definition found for 'com.owncloud.android.presentation.manager.AvatarManager' has been found. Check your module definitions.
        at org.koin.core.scope.Scope.findDefinition(Scope.kt:170)
        at org.koin.core.scope.Scope.resolveInstance(Scope.kt:164)
        at org.koin.core.scope.Scope.get(Scope.kt:128)
        at com.owncloud.android.utils.AvatarUtils$$special$$inlined$inject$1.invoke(Scope.kt:327)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at com.owncloud.android.utils.AvatarUtils.getAvatarManager(Unknown Source:2)
        at com.owncloud.android.utils.AvatarUtils.access$getAvatarManager$p(AvatarUtils.kt:34)
        at com.owncloud.android.utils.AvatarUtils$loadAvatarForAccount$1.invokeSuspend(AvatarUtils.kt:62)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)
2020-07-09 13:53:56.204 18554-18763/? E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-2
    Process: com.owncloud.android, PID: 18554
    org.koin.core.error.NoBeanDefFoundException: No definition found for 'com.owncloud.android.presentation.manager.AvatarManager' has been found. Check your module definitions.
        at org.koin.core.scope.Scope.findDefinition(Scope.kt:170)
        at org.koin.core.scope.Scope.resolveInstance(Scope.kt:164)
        at org.koin.core.scope.Scope.get(Scope.kt:128)
        at com.owncloud.android.utils.AvatarUtils$$special$$inlined$inject$1.invoke(Scope.kt:327)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at com.owncloud.android.utils.AvatarUtils.getAvatarManager(Unknown Source:2)
        at com.owncloud.android.utils.AvatarUtils.access$getAvatarManager$p(AvatarUtils.kt:34)
        at com.owncloud.android.utils.AvatarUtils$loadAvatarForAccount$1.invokeSuspend(AvatarUtils.kt:62)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)
```

The account is stucked after getting this. 

Huaweri P20
Android 9

**Comment by jesmrec** [comment]
The problem (5) that blocked the release, does not happen anymore by building a release apk with latest AS version.

**Comment by jesmrec** [comment]
(5) is not easy to reproduce. We will create new issue to check deeply.

**Comment by jesmrec** [comment]
From my side this is approved and can be moved forward.

**Pull request merged**